core/src/main/java/org/elasticsearch/index/query/functionscore/ExponentialDecayFunctionParser.java

```diff
@@ -33,7 +33,12 @@
     }
 
     @Override
-    public ExponentialDecayFunctionBuilder getBuilderPrototype() {
-        return PROTOTYPE;
+    protected ExponentialDecayFunctionBuilder createFromBytes(String fieldName, BytesReference functionBytes) {
+        return new ExponentialDecayFunctionBuilder(fieldName, functionBytes);
+    }
+
+    @Override
+    public ExponentialDecayFunctionBuilder read(StreamInput t) throws IOException {
```

Comment:
That'd be a big change because that is the method name in Writeable.Reader. Maybe in another PR?